tests/unit/cpp/CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+# The C++ EtcPal tests, all built as one executable or library for now.
+
+etcpal_add_live_test(etcpal_cpp_unit_tests CXX
+  test_error.cpp
+  test_lock.cpp
+  test_main.cpp
+  test_uuid.cpp
+)
```

Comment:
I'm a little unclear how this works with the functions defined in tests/CMakeLists.txt.  Does the function get called multiple times with the TARGET variable bound to the filename?

Reply:
No - functions in CMake are variadic by default. An implicit variable `ARGN` captures a list of any arguments provided past the named ones. This is then used to forward the source file names to `add_executable()` or `add_library()`.